lib/web/app/session.go

```diff
@@ -62,30 +63,34 @@ func (h *Handler) newSession(ctx context.Context, ws types.WebSession) (*session
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	server, err := Match(ctx, accessPoint, MatchPublicAddr(identity.RouteToApp.PublicAddr))
+
+	servers, err := Match(ctx, accessPoint, MatchAll(MatchHealthy(h.c.ProxyClient, identity), MatchPublicAddr(identity.RouteToApp.PublicAddr)))
```

Comment:
Not sure why we need check all app servers healthy status here. 
It looks like the logic in Transport Dial function does already the same thing: https://github.com/gravitational/teleport/pull/9288/files#diff-9dd39725077cc062cbe4ded242810d79af548977d645d288875b70d49d759072R174

Reply:
The idea is to give the transport a more accurate list of servers, and in the case of no healthy status, it would fail before the actual request forward. It is also aligned with the solution, where if the transport's list of servers is empty, the new transport will have only healthy servers to forward requests to, also working as a stop condition of this "renew loop".

Do you think it is better to avoid dialing apps during session creation?

Reply:
Done.